daffodil-core/src/main/java/org/apache/daffodil/api/validation/Validators.java

```diff
@@ -15,49 +15,61 @@
  * limitations under the License.
  */
 
-package org.apache.daffodil.lib.validation
+package org.apache.daffodil.api.validation;
 
-import org.apache.daffodil.lib.api.ValidatorFactory
-import org.apache.daffodil.lib.util.SimpleNamedServiceLoader
+import org.apache.daffodil.util.SimpleNamedServiceLoader;
+
+import java.util.Optional;
 
 /**
- * Access SPI registered [[org.apache.daffodil.lib.api.ValidatorFactory]] instances.
- *
+ * Access SPI registered [[org.apache.daffodil.api.validation.ValidatorFactory]] instances.
+ * <p>
  * Registered instances provide a unique name for lookup.
  */
-object Validators {
-  private lazy val impls: Map[String, ValidatorFactory] = {
-    SimpleNamedServiceLoader.loadClass[ValidatorFactory](classOf[ValidatorFactory])
+public class Validators {
+  private static final Validators INSTANCE = new Validators();
+  private final java.util.Map<String, ValidatorFactory> impls =
+      SimpleNamedServiceLoader.loadClass(ValidatorFactory.class);
+
+  private Validators() {
+  } // private constructor
+
+  public static Validators getInstance() {
```

Comment:
We should use the java singleton syncronization pattern to ensure this doesn't run into any issues with thread safety. I believe the INSTANCE needs to be marked as volatile, and we need a syncronized block.